src/libs/SelectionScraper/index.js

```diff
@@ -0,0 +1,115 @@
+import render from 'dom-serializer';
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import {parseDocument} from 'htmlparser2';
+import _ from 'underscore';
+
+const elementsWillBeSkipped = ['html', 'body'];
+const tagAttribute = 'data-testid';
+
+/**
+ * Reads html of selection. If there is no Selection API returns empty string.
+ * @returns {String} HTML of selection as String
+ */
+const getHTMLOfSelection = () => {
+    if (window.getSelection) {
+        const selection = window.getSelection();
+
+        if (selection.rangeCount > 0) {
+            const div = document.createElement('div');
+
+            // We traverse all ranges, and get closest node with data-testid and replace its contents with contents of
+            // range.
+            for (let i = 0; i < selection.rangeCount; i++) {
+                const range = selection.getRangeAt(i);
+
+                const clonedSelection = range.cloneContents();
+
+                // If clonedSelection has no text content this data has no meaning to us.
+                if (clonedSelection.textContent) {
+                    let node = null;
+
+                    if (range.commonAncestorContainer instanceof HTMLElement) {
+                        node = range.commonAncestorContainer.closest(`[${tagAttribute}]`);
+                    } else {
+                        node = range.commonAncestorContainer.parentNode.closest(`[${tagAttribute}]`);
+                    }
```

Comment:
When is this the case? / What does this else indicate

Reply:
Still looking for this comment, on why the parentNode would have the closest tag attribute instead of the current node